nautobot/docs/guides/graphql.md

```diff
@@ -0,0 +1,1041 @@
+
+```
+query {
+  devices {
+    name
+    interfaces {
+      name
+    }
+  }
+}
+```
+
+The result is a list of all the devices by their hostname and associated interfaces by their names.
+
+![Graph*i*QL - Simple Query 1](/guides/images/graphql/graphiql-06.png)
+
+We can continue iterating on the query until we get exactly what we want from the query. For example, if I wanted to iterate on the previous query to not only display the interfaces of the devices, but also display the interface description, the IP Addresses associated with the interface, and whether or not the interface was a dedicated management interface; I would structure the query like:
+
+```
+query {
+  devices {
+    name
+    interfaces {
+      name
+      description
+      mgmt_only
+      ip_addresses {
+        address
+      }   
+    }
+  }
+}
+```
+
+The results of the query look like:
+
+![Graph*i*QL - Simple Query 2](/guides/images/graphql/graphiql-07.png)
+
+### Filtering Queries
+
+These queries are great, but they are displaying the interface attributes and device names for every device in the Nautobot inventory. Currently, Nautobot allows users to filter queries at the top level of the query. In our previous examples, the top level would be the `devices` query.
+
+As an example. We can query devices by their site location. This is done by added `(site: "<site name>")` after devices. For example: `query { devices(site: "ams") { name }}` will display all devices in the `ams` site.
+
+```
+{
+  "data": {
+    "devices": [
+      {
+        "name": "ams-edge-01"
+      },
+      {
+        "name": "ams-edge-02"
+      },
+      {
+        "name": "ams-leaf-01"
+      },
+      {
+        "name": "ams-leaf-02"
+      },
+      {
+        "name": "ams-leaf-03"
+      },
+      {
+        "name": "ams-leaf-04"
+      },
+      {
+        "name": "ams-leaf-05"
+      },
+      {
+        "name": "ams-leaf-06"
+      },
+      {
+        "name": "ams-leaf-07"
+      },
+      {
+        "name": "ams-leaf-08"
+      }
+    ]
+  }
+}
+```
+
+Graph*i*QL allows you to add multiple attributes to the filter criteria. You can use the `Documentation Explorer` to assist you in finding criteria attributes to filte on. In this example, I add the `manufacturer` attribute in addition to `site`. 
+
+```
+query {
+  devices(site:"ams", manufacturer: "Arista") {
+    name
+  }
+}
+```
+
+```
+{
+  "data": {
+    "devices": [
+      {
+        "name": "ams-edge-01"
+      },
+      {
+        "name": "ams-edge-02"
+      },
+      {
+        "name": "ams-leaf-01"
+      },
+      {
+        "name": "ams-leaf-02"
+      },
+      {
+        "name": "ams-leaf-03"
+      },
+      {
+        "name": "ams-leaf-04"
+      },
+      {
+        "name": "ams-leaf-05"
+      },
+      {
+        "name": "ams-leaf-06"
+      },
+      {
+        "name": "ams-leaf-07"
+      },
+      {
+        "name": "ams-leaf-08"
+      }
+    ]
+  }
+}
+```
```

Comment:
Is the output here supposed to be different than the previous example? It looks the same to me.

Reply:
Changed the query up to produce different results